bika/health/catalog/indexers/patient.py

```diff
@@ -21,14 +21,7 @@ def listing_searchable_text(instance):
     columns = catalog.schema()
 
     for column in columns:
-        try:
-            value = api.safe_getattr(instance, column)
-        except:
-            logger.debug("{} has no attribute called '{}' ".format(
-                            repr(instance), column))
-            continue
-        if not value:
-            continue
+        value = api.safe_getattr(instance, column, None)
```

Comment:
I would really like to get the column values of the brain here and not of the instance. IMO this should be easy by doing this:

```python
brain = None
at_tool = api.get_tool("archetype_tool")
catalog = at_tool.getCatalogsByType(instance.portal_type)
if len(catalog) > 0:
    brain = catalog[0]({"UID": api.get_uid(instance)})
brain_value = api.safe_getattr(brain, column, None)
instance_value = api.safe_getattr(instance, column, None)
parsed = api.to_searchable_text_metadata(brain_value or instance_value)
entries.append.parsed()
```

Reply:
Also see how the `ReportModel` of the publisher is doing that: https://github.com/senaite/senaite.publisher/blob/master/src/senaite/publisher/reportmodel.py#L204

Reply:
That would be great and safer, but I think since this is a method to get index value for the instance, it means this method will be called while creating/building a brain for the instance in the catalog. I am not sure if we can access to that catalog by the time we are building it, but even if we do, nothing will be found for the instance's UID, because the instance has not been 'cataloged' yet.